packages/web/integrations/notifi/notifi-subscription-card/notifi-subscription-card.tsx

```diff
@@ -11,17 +11,28 @@ import { ExpiredCard } from "~/integrations/notifi/notifi-subscription-card/expi
 import { FetchedCard } from "~/integrations/notifi/notifi-subscription-card/fetched-card";
 import { LoadingCard } from "~/integrations/notifi/notifi-subscription-card/loading-card";
 
-export const NotifiSubscriptionCard: FunctionComponent = () => {
-  const { setIsOverLayEnabled, renderView } = useNotifiModalContext();
+type Props = {
+  parentType?: "popover" | "modalBase";
+};
+
+export const NotifiSubscriptionCard: FunctionComponent<Props> = ({
+  parentType,
+}) => {
+  const { setIsOverLayEnabled, renderView, setIsCardOpen } =
+    useNotifiModalContext();
 
   const { client } = useNotifiClientContext();
 
   const config = useNotifiConfig();
 
-  const { cardView, setCardView, setEmail, setTelegramId, setPhoneNumber } =
+  const { cardView, setEmail, setTelegramId, setPhoneNumber } =
     useNotifiSubscriptionContext();
   const firstLoadRef = useRef(false);
 
+  useEffect(() => {
+    setIsCardOpen(parentType ? true : false);
+  }, [parentType]);
```

Comment:
Is it possible to remove this useEffect? Instead can we derive more state from `parentType`?

Reply:
I understand this part is sort of confusing because it is a work-around of `@reactheadlessui`'s `Popover` component. In order to implement the discard/save modal, we need to manually handle the `Popover`'s `open` state instead of using it's interal state to open or close the Popover component.

The `discard/save modal` can shows only when the internal state (`popOverOpen`) of `Popover` is `flase` and the `isPreventingCardClosed` is `true`. 

To do so, we need to know everytime when the `popOverOpen` state changes. The `useEffect` is needed so that `Card` component is able to get the internal state when it is changed. 

But let me try to make a refactor PR try to make it more difficult to follow.
CC: @jonator : Hope it makes sense to you